uECC.c

```diff
@@ -172,8 +172,11 @@ static cmpresult_t uECC_vli_cmp_unsafe(const uECC_word_t *left,
                                        const uECC_word_t *right,
                                        wordcount_t num_words);
 
-#if (uECC_PLATFORM == uECC_arm || uECC_PLATFORM == uECC_arm_thumb || \
-        uECC_PLATFORM == uECC_arm_thumb2)
+#if defined( __arm__ ) && !defined( __ARM_ARCH_7A__ ) && !defined( __ARM_ARCH_7R__ ) && !defined( __ARM_ARCH_7M__ ) && !defined( __ARM_ARCH_7S__ )
```

Comment:
Actually, the better fix for Android armeabi is just to build `uECC.c` with `-marm` option and make no changes to the source code at all.

By building with `-marm` option (default is thumb mode), assembler outputs this message:

    rdhi, rdlo and rm must all be different

but the binary works correctly.